test/session_test/password_test.rb

```diff
@@ -86,6 +87,33 @@ def test_credentials
         assert_equal({ :password => "<protected>", :login => "login" }, session.credentials)
       end
 
+      def test_action_controller_parameters_credentials
+        session = UserSession.new
+        session.credentials = ActionController::Parameters.new(
+          user: {
+            login: 'login',
+            password: 'pass'
+          }
+        ).require(:user).permit(:login, :password)
+
+        assert_equal "login", session.login
+        assert_nil session.password
+        assert_equal "pass", session.send(:protected_password)
+        assert_equal({ :password => "<protected>", :login => "login" }, session.credentials)
+      end
+
+      def test_non_permitted_action_controller_parameters_credentials
+        session = UserSession.new
+        assert_raise(ArgumentError) do
+          session.credentials = ActionController::Parameters.new(
+            user: {
+              login: 'login',
+              password: 'pass'
+            }
+          ).require(:user)
```

Comment:
Please also test the error message, asserting that it mentions which keys were missing.